src/mixins/createRenderMixin.ts

```diff
@@ -1,254 +1,9 @@
-				const props: VNodeProperties = {};
-
-				this.nodeAttributes.forEach((fn) => {
-					const newProps: VNodeProperties = fn.call(this, assign({}, props));
-					if (newProps) {
-						assign(props, newProps);
-					}
-				});
-
-				if (overrides) {
-					assign(props, overrides);
-				}
-				return props;
-			},
-
-			getSelectorAndWidgetClasses(this: RenderMixin<RenderMixinState>): string {
-				const selectorAndClasses = [this.tagName, ...this.classes];
-				return selectorAndClasses.join('.');
-			},
-
-			getChildrenNodes(this: RenderMixin<RenderMixinState>): (VNode | string)[] {
-				return this.state.label ? [ this.state.label ] : [];
-			},
-
-			get id(this: RenderMixin<RenderMixinState>): string {
-				return (this.state && this.state.id) || generateID(this);
-			},
-
-			invalidate(this: RenderMixin<RenderMixinState>): void {
-				if (dirtyMap.get(this)) { /* short circuit if already dirty */
-					return;
-				}
-				const parent = this.parent;
-				dirtyMap.set(this, true);
-				renderCache.delete(this); /* Allow GC to occur on renderCache */
-				if (parent && parent.invalidate) {
-					parent.invalidate();
-				}
-			},
-
-			nodeAttributes: [
-				function (this: RenderMixin<RenderMixinState>): VNodeProperties {
-					const baseIdProp = this.state && this.state.id ? { 'data-widget-id': this.state.id } : {};
-					const { styles = {} } = this.state;
-					const classes: { [index: string]: boolean; } = {};
-					const widgetClasses = widgetClassesMap.get(this);
-
-					widgetClasses.forEach((c) => classes[c] = false);
-
-					if (this.state && this.state.classes) {
-						this.state.classes.forEach((c) => classes[c] = true);
-						widgetClassesMap.set(this, this.state.classes);
-					}
-
-					return assign(baseIdProp, { key: this, classes, styles });
-				}
-			],
-
-			parent: null,
-
-			render(this: RenderMixin<RenderMixinState>): VNode {
-				const cachedRender: RenderMixin<RenderMixinState> = this;
-				let cached = renderCache.get(cachedRender);
-				if (!dirtyMap.get(cachedRender) && cached) {
-					return cached;
-				}
-				else {
-					cached = h(cachedRender.getSelectorAndWidgetClasses() , cachedRender.getNodeAttributes(), cachedRender.getChildrenNodes());
-					renderCache.set(cachedRender, cached);
-					dirtyMap.set(cachedRender, false);
-					return cached;
-				}
-			},
-
-			classes: [],
-
-			tagName: 'div'
-		},
-		initialize(instance: RenderMixin<RenderMixinState>, options: RenderMixinOptions<RenderMixinState> = {}) {
-			const { tagName, render, parent } = options;
-			instance.tagName = tagName || instance.tagName;
-			instance.render = render || instance.render;
-
-			if (parent) {
-				parent.append(instance);
-			}
-
-			dirtyMap.set(instance, true);
-
-			instance.own(instance.on('state:changed', () => instance.invalidate()));
-
-			widgetClassesMap.set(instance, []);
-		}
-	})
-	.static({
-		idBase: 'widget'
-	}) as RenderMixinFactory;
+export {
```

Comment:
I assume the main reason we are doing this is to minimise the amount of refactoring?